tfhe/src/integer/server_key/radix_parallel/tests.rs

```diff
@@ -834,25 +834,19 @@ fn integer_unchecked_scalar_left_shift(param: Parameters) {
     // message_modulus^vec_length
     let modulus = param.message_modulus.0.pow(NB_CTXT as u32) as u64;
 
-    //Nb of bits to shift
     let tmp_f64 = param.message_modulus.0 as f64;
     let nb_bits = tmp_f64.log2().floor() as usize * NB_CTXT;
 
     for _ in 0..NB_TEST {
         let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u64>() % nb_bits as u64;
```

Comment:
don't we want to check the overflow behavior ? We can do checked shift on the clear values

same thing for all tests on shifts that have the same pattern

Reply:
Yes we should have test for when the input shift is > nb_bits.

the current behavior is that implemented is (should be)  equivalent to `x.checked_shl(y).unwrap_or(0)`, that is, if the shift is bigger than nb_bits the bits are discarded. Which is what I'm accustomed to.

default for rust is a bit different, something like `3u8 << 232` won't compile in debug mode with 
```
 attempt to shift left by `232_i32`, which would overflow
```

In release mode it compiles and gets replaced with something like `3u8 << (232 & 7)` which as en quivialent
`3u8.wrapping_shl(232)`.

What default should be chose ?

Reply:
I would expect an overflowing shift to output 0 I think, I think that as long as it's well documented it should be fine